### PR TITLE
custom fields

### PR DESCRIPTION
**Pull request by hunar1312**

### CODE REVIEW

File: app/components/assets/form.tsx [hunk 1]
```diff
@@ -53,6 +54,21 @@ export const AssetForm = ({
   const [, validateFile] = useAtom(validateFileAtom);
   const [, updateTitle] = useAtom(updateTitleAtom);
 
+  const fields = [
```
Comment (DonKoko) [inline app/components/assets/form.tsx]: We should not forget to clean this up. Just leaving a note.

File: app/components/assets/form.tsx [hunk 2]
```diff
@@ -163,7 +180,51 @@ export const AssetForm = ({
...
+              key={field.id}
+              rowLabel={field.name}
+              subHeading={field.helpText && <p>{field.helpText}</p>}
+              className="border-b-0"
+            >
+              <Input
+                hideLabel
+                type={field.type}
+                label={field.name}
+                name={field.name}
```
Comment (DonKoko) [inline app/components/assets/form.tsx]: We are missing the required. Also a bit of a challenge. We have some schema for creating/editing assets form data. We need to see how we can dynamically add to the schema based on the custom fields config.

File: app/components/assets/form.tsx [hunk 3]
```diff
@@ -163,7 +180,51 @@ export const AssetForm = ({
...
+                className="w-full"
+              />
+            </FormRow>
+          ))
+        ) : (
+          <div>
```
Comment (DonKoko) [inline app/components/assets/form.tsx]: Lets move this to its own component so we can keep the file cleaner. Its already quite big.

File: app/components/custom-fields/form.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+interface Props {
+  title?: string;
+  helpText?: string;
+}
+
+export const CustomFieldForm = ({ title, helpText }: Props) => {
+  const navigation = useNavigation();
+  const zo = useZorm("NewQuestionWizardScreen", NewCustomFieldFormSchema);
+  const disabled = isFormProcessing(navigation.state);
+  const fieldTypes = ["Text"];
```
Comment (DonKoko) [inline app/components/custom-fields/form.tsx]: This will need to be cleaned up.

Comment (hunar1312) [reply app/components/custom-fields/form.tsx]: yes, this all static content will be cleaned up. This was added just to make frontend work and backend integration easier

File: app/components/assets/form.tsx (hunk 3 above)
Comment (hunar1312) [reply app/components/assets/form.tsx]: sure

File: app/components/custom-fields/form.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+  const [, updateTitle] = useAtom(updateTitleAtom);
+
+  const organizationId = useOrganizationId();
+  return (
+    <Form
```
Comment (DonKoko) [inline app/components/custom-fields/form.tsx:63]: Submitting with cmd+enter doesn't seem to work and it works on example on asset.new. Can you please check why?

File: app/routes/_layout+/settings.custom-fields.index.tsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    totalPages,
+    perPage,
+    prev,
+    next,
+    modelName,
+  });
+}
+
+export const action = async ({ request }: ActionArgs) => {
+  await requireAuthSession(request);
```
Comment (DonKoko) [inline app/routes/_layout+/settings.custom-fields.index.tsx:74]: Here we only have a delete action so I would use the `isDelete()` helper to make sure the request has the correct method.

**Review by DonKoko — CHANGES_REQUESTED** [review]
Overall really good. We discussed some stuff on discord and I also added some comments here. What is also missing the edit view.

**Review by DonKoko — CHANGES_REQUESTED** [review]
Clicking on the actions dropdown of the custom field, navigates to the field page. Not sure why it happens

**Pull request merged**